Add final rewards+final used fee

johnnyji-dev opened the pull request.

johnnyji-dev committed:
add some api(claimedReward, UsedFee).
- After completing transaction, need to retrieve a final claimed reward and used fee.

johnnyji-dev committed:
update claimedReward & usedFee param (address to extrinsicHash)

gtg7784 commented:
Hi @johnnyji-dev thank you for your contribution. Can you please add the PR dscription?

johnnyji-dev committed:
update claimedReward(Claim_staker_rewards & Claim_bonus_reward)

johnnyji-dev commented:
> Hi @johnnyji-dev thank you for your contribution. Can you please add the PR dscription?

Is there any template for PR description? Is there no format?
Can I just write down description for my PR?
BTW, I checked out my PR and it need to update on "claimedReward" method.

I completed to update and test that method.
I am going to retry to PR.

gtg7784 commented:
> > Hi @johnnyji-dev thank you for your contribution. Can you please add the PR dscription?
> 
> Is there any template for PR description? Is there no format? Can I just write down description for my PR? BTW, I checked out my PR and it need to update on "claimedReward" method.
> 
> I completed to update and test that method. I am going to retry to PR.

Hi, you can write summary of updates in this PR without template.

Please update the description and check that claimedReward method (and why it's needed)

thank you for your contribution.

gtg7784 commented:
and also, please run lint before commit your updates. thank you

johnnyTGV commented:
> and also, please run lint before commit your updates. thank you

Okay~ After work, I will do that~

bobo-k2 commented:
Hi @johnnyji-dev on more thing. Can you please uplift version in `package.json` (there are multiple files) from `0.2.9` to `0.3.0`

johnnyji-dev committed:
fix: version change 0.2.8 => 0.3.0 on package.json

johnnyji-dev commented:
> Hi @johnnyji-dev on more thing. Can you please uplift version in `package.json` (there are multiple files) from `0.2.9` to `0.3.0`

![Monosnap package json — astar js  개발 컨테이너: Alpine @ desktop-linux  2024-06-23 16-56-06](https://github.com/AstarNetwork/astar.js/assets/125259310/0387f188-c269-45fd-a118-f271642a72ba)

I uplifted version in package.json (there are multiple files) to 0.3.0

johnnyji-dev committed:
Merge branch 'main' into add_FinalRewards+FinalUsedFee

impelcrypto reviewed (changes requested):
Hi @johnnyji-dev 
Thank you for your contribution. Could you help update the version number to 0.3.0 (there are required updating in multiple package.json files)? Thanks in advance.

Ref:
https://github.com/AstarNetwork/astar.js/pull/69/files

bobo-k2 commented on packages/sdk-core/src/modules/dapp-staking/pending-rewards/index.ts:
```diff
@@ -251,3 +251,128 @@ export const estimatePendingRewards = async ({ api,
+    let usedFee = BigInt('0');
+
+    extrinsicEvents.map((e) => {
+      if (e.toHuman()?.event?.method == method &&
```
Can you avoid using `toHuman` and use polkadot.js types instead. This will also remove need for `commaStrToBigInt`
The same applies for the method above

impelcrypto commented:
Could you run `yarn lint`? CI failed in `lint` and `build` .

johnnyji-dev commented:
After work, I will check them out. Thx~

2024년 7월 8일 (월) 오후 5:22, Roy ***@***.***>님이 작성:

> Could you run yarn lint? CI failed in lint and build .
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AstarNetwork/astar.js/pull/75#issuecomment-2213342472>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A53U4LRLMGZCEN2RX4A3RUTZLJD27AVCNFSM6AAAAABELFDJAWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDEMJTGM2DENBXGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

johnnyji-dev committed:
fix : yarn lint - modification

johnnyji-dev committed:
fix : remove toHuman() & commaStrToBigInt()

bobo-k2 reviewed (changes requested):
Hi @johnnyji-dev, please address the following
- fix build errors
- provide PR description and explain what you did and why those changes are needed
- Update `SDK-Core Library Functions` section in `README.md` 
- Have in mind that there are actually three different reward events in case you want to also handle the third one https://github.com/AstarNetwork/Astar/blob/150b11c4ffabaf82f6c1ec165bfa5d99340c8358/pallets/dapp-staking-v3/src/lib.rs#L301

johnnyji-dev committed:
fix : add DAppReward on claimedReward & update README.md

johnnyji-dev committed:
fix : update UsedFee() to getUsedFee()

johnnyji-dev committed:
fix : update UsedFee() to getUsedFee() on README.md

johnnyji-dev committed:
fix : Update claimedReward to getClaimedReward

johnnyji-dev commented:
> Hi @johnnyji-dev, please address the following
> 
> * fix build errors
> * provide PR description and explain what you did and why those changes are needed
> * Update `SDK-Core Library Functions` section in `README.md`
> * Have in mind that there are actually three different reward events in case you want to also handle the third one https://github.com/AstarNetwork/Astar/blob/150b11c4ffabaf82f6c1ec165bfa5d99340c8358/pallets/dapp-staking-v3/src/lib.rs#L301

- Updated method name (claimedReward to getClaimedReward & UsedFee to getUsedFee)
- Updated README.md file ( add getClaimedReward() & getUsedFee() description. )
- Add "Dapp Reward" case on getClaimedReward()

bobo-k2 commented:
> > Hi @johnnyji-dev, please address the following
> > 
> > * fix build errors
> > * provide PR description and explain what you did and why those changes are needed
> > * Update `SDK-Core Library Functions` section in `README.md`
> > * Have in mind that there are actually three different reward events in case you want to also handle the third one https://github.com/AstarNetwork/Astar/blob/150b11c4ffabaf82f6c1ec165bfa5d99340c8358/pallets/dapp-staking-v3/src/lib.rs#L301
> 
> * Updated method name (claimedReward to getClaimedReward & UsedFee to getUsedFee)
> * Updated README.md file ( add getClaimedReward() & getUsedFee() description. )
> * Add "Dapp Reward" case on getClaimedReward()

Hi @johnnyji-dev you forgot the most important point. The code is not building
<img width="1027" alt="image" src="https://github.com/user-attachments/assets/c46fd3ef-a43c-47d6-9c28-b82b1d563821">

johnnyji-dev committed:
fix : update build-error(yarn polkadot-exec-tsc --build tsconfig.build.json)

johnnyji-dev commented:
> > > Hi @johnnyji-dev, please address the following
> > > 
> > > * fix build errors
> > > * provide PR description and explain what you did and why those changes are needed
> > > * Update `SDK-Core Library Functions` section in `README.md`
> > > * Have in mind that there are actually three different reward events in case you want to also handle the third one https://github.com/AstarNetwork/Astar/blob/150b11c4ffabaf82f6c1ec165bfa5d99340c8358/pallets/dapp-staking-v3/src/lib.rs#L301
> > 
> > 
> > 
> > * Updated method name (claimedReward to getClaimedReward & UsedFee to getUsedFee)
> > * Updated README.md file ( add getClaimedReward() & getUsedFee() description. )
> > * Add "Dapp Reward" case on getClaimedReward()
> 
> Hi @johnnyji-dev you forgot the most important point. The code is not building <img alt="image" width="1027" src="https://private-user-images.githubusercontent.com/8452361/350921020-c46fd3ef-a43c-47d6-9c28-b82b1d563821.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE2NjA4MjQsIm5iZiI6MTcyMTY2MDUyNCwicGF0aCI6Ii84NDUyMzYxLzM1MDkyMTAyMC1jNDZmZDNlZi1hNDNjLTQ3ZDYtOWMyOC1iODJiMWQ1NjM4MjEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjJUMTUwMjA0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDY1OTg4Y2MwNDIwNGE3MzFjY2JjMzY3OTliMTExYmRjYWE5MmIxZTI3OWVkYjBlOGM0NzQ2NmRhZjEyZGExNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.EE6k7r0OWtvkHR_E81k4QQvhMQDlgxD_Cxxzdeotx9Q">

Oh, I missed that part!! And complete to update it.

bobo-k2 commented on packages/sdk-core/src/modules/dapp-staking/pending-rewards/index.ts:
```diff
@@ -321,7 +322,7 @@ export const getUsedFee = async (
     const signedBlock = await api.rpc.chain.getBlock(blockHash);
     const apiAt = await api.at(signedBlock.block.header.hash);
-    const allRecords = (await apiAt.query.system.events()).toArray();
+    const allRecords: any = await apiAt.query.system.events();
```
Please don't use `any` Applicable to multiple locations within the code
https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html

johnnyji-dev replied:
@bobo-k2 
BTW, Property 'map' does not exist on type 'Codec'.
The type of 'allRecords' is Codec.
I tried to iterate the data by using 'filter' or 'map'. But that make build error(yarn polkadot-exec-tsc --build tsconfig.build.json).

How can I handle this? Can you suggest any reference docs?

bobo-k2 replied:
> @bobo-k2 BTW, Property 'map' does not exist on type 'Codec'. The type of 'allRecords' is Codec. I tried to iterate the data by using 'filter' or 'map'. But that make build error(yarn polkadot-exec-tsc --build tsconfig.build.json).
> 
> How can I handle this? Can you suggest any reference docs?

Codec is a base type from which all polkadot.js types are inherited. `System.events` are of type `Vec<FrameSystemEventRecord>` so you can change your command to
```
const allRecords = await api.query.system.events<Vec<FrameSystemEventRecord>>();
```

bobo-k2 replied:
Codec is a base type from which all polkadot.js types are inherited. `System.events` are of type `Vec<FrameSystemEventRecord>` so you can change your line to
```
const allRecords = await api.query.system.events<Vec<FrameSystemEventRecord>>();
```

johnnyji-dev replied:
When '<Vec<FrameSystemEventRecord>>' applied, can not import the member like below error message. 
packages/sdk-core/src/modules/dapp-staking/pending-rewards/index.ts:7:10 - error TS2305: Module '"@polkadot/types/lookup"' has no exported member 'FrameSystemEventRecord'.

Alternative, I used the 'EventRecord' and got properties(actualFee, amount) by index.

bobo-k2 reviewed (changes requested):
Still missing the PR description. Explain reasons for the change

johnnyji-dev committed:
fix : apply <Vec<EventRecord>> on system.events & delete any

johnnyji-dev committed:
fix : delete the unusing comments
